pysrc/bytewax/inputs.py

```diff
@@ -13,6 +13,69 @@
 from .bytewax import AdvanceTo, Emit
 
 
+def distribute(elements: Iterable[Any], index: int, count: int) -> Iterable[Any]:
+    """Distribute elements equally between a number of buckets and return
+    the items for the given bucket index.
+
+    No two buckets will get the same element.
+
+    >>> list(distribute(["blue", "green", "red"], 0, 2))
+    ['blue', 'red']
+    >>> list(distribute(["blue", "green", "red"], 1, 2))
+    ['green']
+
+    Note that if you have more buckets than elements, some buckets
+    will get nothing.
+
+    >>> list(distribute(["blue", "green", "red"], 3, 5))
+    []
+
+    This is very useful when writing input builders and you want each
+    of your workers to handle reading a disjoint partition of your
+    input.
+
+    >>> from bytewax import Dataflow, spawn_cluster
+    >>> def read_topics(topics):
+    ...     for topic in topics:
+    ...         for i in enumerate(3):
+    ...             yield f"topic:{topic} item:{i}"
+    >>> def input_builder(i, n):
+    ...    all_topics = ["red", "green", "blue"]
+    ...    this_workers_topics = distribute(listening_topics, i, n)
+    ...    for item in read_topics(this_workers):
+    ...        yield Emit(f"worker_index:{worker_index}" + item)
```

Comment:
I wanted to avoid as much talk of epochs here because this function doesn't deal with them or help you craft a specific interpretation of them. Although one might argue that "having everything in one batch in the default epoch" is a strange assumption for the example too.